cloud/scope/machine.go

```diff
@@ -767,6 +772,75 @@ func (m *MachineScope) DeleteVPCLoadBalancerPoolMember() error {
+func (m *MachineScope) deleteVPCLoadBalancerPoolMembers() error {
+	for _, member := range m.IBMVPCMachine.Status.LoadBalancerPoolMembers {
+		// Retrieve the Instance details immediately (without them the member cannot be safely deleted).
+		instanceOptions := &vpcv1.GetInstanceOptions{
+			ID: ptr.To(m.IBMVPCMachine.Status.InstanceID),
+		}
+		instanceDetails, _, err := m.IBMVPCClient.GetInstance(instanceOptions)
+		if err != nil {
+			return fmt.Errorf("error retrieving instance for load balancer pool member deletion for machine %s: %w", m.IBMVPCMachine.Name, err)
+		}
+		// Verify the instance has a primary network interface IP address needed to delete the correct Load Balancer Pool Member.
+		if instanceDetails.PrimaryNetworkInterface == nil || instanceDetails.PrimaryNetworkInterface.PrimaryIP == nil || instanceDetails.PrimaryNetworkInterface.PrimaryIP.Address == nil {
+			return fmt.Errorf("error instance is missing the primary network interface IP address for load balancer pool member deletion for machine: %s", m.IBMVPCMachine.Name)
+		}
+		m.Logger.V(5).Info("collected instance details for load balancer pool member deletion", "machienName", m.IBMVPCMachine.Name, "instanceID", *instanceDetails.ID, "instanceIP", *instanceDetails.PrimaryNetworkInterface.PrimaryIP.Address)
+
+		// Retrieve the Load Balancer ID.
+		loadBalancerID, err := m.getLoadBalancerID(ptr.To(member.LoadBalancer))
+		if err != nil {
+			return fmt.Errorf("error retrieving load balancer id for load balancer pool member deletion for machine %s: %w", m.IBMVPCMachine.Name, err)
+		}
+		m.Logger.V(5).Info("collected load balancer id for load balancer pool member deletion", "machineName", m.IBMVPCMachine.Name, "loadBalancerID", *loadBalancerID)
+
+		// Lookup the Load Balancer Pool ID, if only name is available.
+		loadBalancerPoolID, err := m.getLoadBalancerPoolID(ptr.To(member.Pool), *loadBalancerID)
+		if err != nil {
+			return fmt.Errorf("error retrieving load balancer pool id for load balancer pool member deletion for machine %s: %w", m.IBMVPCMachine.Name, err)
+		}
+		m.Logger.V(5).Info("collected load balancer pool id for load balancer pool member deletion", "machineName", m.IBMVPCMachine.Name, "loadBalancerPoolID", *loadBalancerPoolID)
+
+		listMembersOptions := &vpcv1.ListLoadBalancerPoolMembersOptions{
+			LoadBalancerID: loadBalancerID,
+			PoolID:         loadBalancerPoolID,
+		}
+
+		m.Logger.V(5).Info("list load balancer pool members options", "machineName", m.IBMVPCMachine.Name, "options", *listMembersOptions)
+		poolMembers, _, err := m.IBMVPCClient.ListLoadBalancerPoolMembers(listMembersOptions)
+		if err != nil {
+			return fmt.Errorf("error retrieving load balancer pool members for load balancer pool member deletion for machine %s: %w", m.IBMVPCMachine.Name, err)
+		}
+
+		for _, poolMember := range poolMembers.Members {
+			if poolMemberTarget, ok := poolMember.Target.(*vpcv1.LoadBalancerPoolMemberTarget); ok {
```

Comment:
May be if we negate the condition it looks cleaner.
```
if poolMemberTarget, ok := poolMember.Target.(*vpcv1.LoadBalancerPoolMemberTarget); !ok {
    continue
}
```

Reply:
I can add a continue, but I cannot make the change you are asking for, as `poolMemberTarget` will only be scoped to the if block, which will now only contain `continue`.

Reply:
Sure, I'll change the API call for the LB to adjust how the error gets reported for the LB status.